### ISSUE TITLE
[Bug]: Running Next Dev locally with MiddlewareRequest causes error

### ISSUE THREAD

**Issue by isaac-martin** [netlify/next-runtime#2003]
### Summary

Running `next dev` locally with `MiddlewareRequest` inside middleware.ts causes an error `MiddlewareRequest only works in a Netlify Edge Function environment`

<img width="1162" alt="image" src="https://user-images.githubusercontent.com/8649366/226705572-5b3a2a7d-096b-4318-b2d9-22b8be05f290.png">

If you have to run locally using `netlify dev` this should be called out inside the documentation. 

Subsequently there is a typescript error also ` Types of property 'geo' are incompatible.`

### Steps to reproduce

Inside middleware.ts `import { MiddlewareRequest } from "@netlify/next";`
create a new netlify middleware request with the below code
`const request = new MiddlewareRequest(req);`
 go to a url in browser and notice the error

### Next Runtime version

x

### Is your issue related to the `app` directory (beta)?

- [ ] Yes, I am using the `app` directory

### More information about your build

- [ ] I am building using the CLI
- [ ] I am building using file-based configuration (`netlify.toml`)

### What OS are you using?

Mac OS

### Your netlify.toml file

<details>
  <summary>`netlify.toml`</summary>

```toml
# Paste content of your `netlify.toml` file here
```

</details>

### Your public/_redirects file

<details>
  <summary>`_redirects`</summary>

```
# Paste content of your `_redirects` file here
```

</details>

### Your `next.config.js` file

<details>
  <summary>`next.config.js`</summary>

```js
# Paste content of your `next.config.js` file here. Check there is no private info in there.
```

</details>

### Builds logs (or link to your logs)

<details>
  <summary>Build logs</summary>

```
# Paste logs here
```

</details>

### Function logs

<details>
  <summary>Function logs</summary>

```
# Paste logs here
```

</details>

### .next JSON files

<details>
  <summary>generated .next JSON files</summary>

```
# Paste file contents here. Please check there isn't any private info in them
# You can either build locally, or download the deploy from Netlify by clicking the arrow next to the deploy time.
```

</details>

**Comment by sarahetter** [comment]
Closing as stale

**Comment by isaac-martin** [comment]
Sorry @sarahetter have been off work for the last few months. Am i correct in assuming that you are running your local dev environment with `netlify dev` ? Is this a requirement for using this?

**Comment by sarahetter** [comment]
Yes, I was running with `netlify dev`. The `@netlify/next` package makes use of edge functions, and so using our `dev` command rather than `next dev` is required. 

I'll create an issue to make this clearer in the documentation.

**Comment by pdme** [comment]
I am experiencing the same issue, even when running `ntl dev`.

To reproduce, I followed the steps above. I created a new next project (tried with several different options, app/pages, src, etc), installed `@netlify/next`, added the `middleware.ts` just like above, and ran `ntl dev`.

Then I get:
```
Error: MiddlewareRequest only works in a Netlify Edge Function environment
    at middleware (middleware.ts:6:18)
  4 | export async function middleware(req: NextRequest) {
  5 |
> 6 |   const request = new MiddlewareRequest(req)
    |                  ^
  7 |   
  8 |   return request.next()
  9 | }
```

Can we please reopen this issue?
